### PR TITLE
#175  Passport

### PR DESCRIPTION
**Pull request by hunterwilliams**
#175 Using as a placeholder. Also so anyone can take a peek if they have any tips. I have more of the development locally... but avoiding committing because a lot I think I'll have to back out

@joemfb - know you were interested

### CODE REVIEW

**Comment by hunterwilliams** [comment]
Serverside is done and works as previously except possible weirdness with updating the user, but it could have been odd before.

File: app/templates/node-server/routes.js [hunk 1]
```diff
@@ -4,100 +4,43 @@
...
+    user = req.session.passport.user;
+  }
+  else if (req.session.user)
+  {
+    user = req.session.user;
+  }
+  if (!user) {
     res.send('{"authenticated": false}');
   } else {
+    var name = user.username || user.name;
```
Comment (hunterwilliams) [inline app/templates/node-server/routes.js]: this whole method may just need some cleanup

**Comment by hunterwilliams** [comment]
Looks like I can swap out the redirect and basically make the login service back to what it was. Just need to implement this -> http://toon.io/on-passportjs-specific-use-cases/ and use a different ``passport.authenticate`` method to return json like it used to.

I'll do this later. If you have any input @joemfb on that or any other thoughts before I do this then prettify it - mostly pertaining to allow it to support other logins easily.

**Comment by hunterwilliams** [comment]
Managed to get it working as said without changes to ui. However, it'd be significantly harder to drag and drop in a new login strategy. So going to try to make the ui adapt to the server changes instead.

File: app/templates/node-server/utils/auth.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            if (json.user !== undefined) {
+              user.profile = json.user;
+            } else {
+              console.log('did not find chunk.user');
+            }
+
+            done(null, user);
+          });
+        } else if (response.statusCode === 401) {
+          done(null, false, {message: 'Invalid credentials'});
```
Comment (hunterwilliams) [inline app/templates/node-server/utils/auth.js]: Actually may need to change this to saying logged in just has no user profile..

**Comment by hunterwilliams** [comment]
Fixed a bug with it not allowing logins without existing profiles. Got it working with the current master. So it's ready to be merged.

**Comment by grtjn** [comment]
Looks like there is a lot of good stuff in this PR. Anyone more familiar with Passport and such than me that likes to take a look?

**Comment by grtjn** [comment]
These Passport changes were (manually) merged into PR #285, closing this one. Thanks Hunter for your work!

**Comment by hunterwilliams** [comment]
Haha that's awesome! Can't imagine it was a fun merge

**Pull request closed without merging**